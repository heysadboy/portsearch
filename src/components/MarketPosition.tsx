import { FC } from "react";

interface IMarketPositionProp {
    setSelectedPositions: (selectedPositions: string[]) => void
    selectedPositions: string[]
}

const MarketPosition: FC<IMarketPositionProp> = ({ setSelectedPositions, selectedPositions }) => {

    const setMarketPosition = (e: any) => {
        let positions = [...selectedPositions];
        switch (e.target.name) {
            case "high":
                e.target.checked ? positions.push("high") : positions = positions.filter(position => position !== "high")
                break;
            case "average":
                e.target.checked ? positions.push("average") : positions = positions.filter(position => position !== "average")
                break;
            case "low":
                e.target.checked ? positions.push("low") : positions = positions.filter(position => position !== "low")
                break;
        }
        setSelectedPositions(positions);
    }

    return (
        <div className="grouped fields">
            <div className="market-position-title">Market Position</div>
            <div className="field">
                <div className="ui checkbox market-checkbox">
                    <input type="checkbox" name="high" onChange={setMarketPosition} />
                    <label>High</label>
                </div>
            </div>
            <div className="field">
                <div className="ui checkbox market-checkbox">
                    <input type="checkbox" name="average" defaultChecked={true} onChange={setMarketPosition} />
                    <label>Average</label>
                </div>
            </div>
            <div className="field">
                <div className="ui checkbox market-checkbox">
                    <input type="checkbox" name="low" onChange={setMarketPosition} />
                    <label>Low</label>
                </div>
            </div>
        </div>
    );
}

export default MarketPosition;