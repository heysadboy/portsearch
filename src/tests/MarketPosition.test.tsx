import TestRenderer from "react-test-renderer";
import { act } from 'react-dom/test-utils';
import MarketPosition from "../components/MarketPosition";
import ReactDOM from "react-dom";

let selectedPositions: string[] = [];
const setSelectedPositions = (data: string[]) => {
    selectedPositions = [...selectedPositions, ...data]
}

test("MarketPosition Component Testing.", () => {
    const component = TestRenderer.create(
        <MarketPosition selectedPositions={selectedPositions} setSelectedPositions={setSelectedPositions} />
    )

    let tree: any = component.toJSON();
    expect(tree).toMatchSnapshot();
});

let container: any;

beforeEach(() => {
    container = document.createElement('div');
    document.body.appendChild(container);
});

afterEach(() => {
    document.body.removeChild(container);
    container = null;
});

it('Click the checkbox to see if state is updated.', () => {
    act(() => {
        ReactDOM.render(<MarketPosition selectedPositions={selectedPositions} setSelectedPositions={setSelectedPositions} />, container);
    });

    const input = container.querySelector("input")
    act(() => {
        input.dispatchEvent(new MouseEvent('click', { bubbles: true }));
    });
    expect(selectedPositions).toEqual(["high"]);
});