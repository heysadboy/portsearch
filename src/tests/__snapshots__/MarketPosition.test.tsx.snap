// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MarketPosition Component Testing. 1`] = `
<div
  className="grouped fields"
>
  <div
    className="market-position-title"
  >
    Market Position
  </div>
  <div
    className="field"
  >
    <div
      className="ui checkbox market-checkbox"
    >
      <input
        name="high"
        onChange={[Function]}
        type="checkbox"
      />
      <label>
        High
      </label>
    </div>
  </div>
  <div
    className="field"
  >
    <div
      className="ui checkbox market-checkbox"
    >
      <input
        defaultChecked={true}
        name="average"
        onChange={[Function]}
        type="checkbox"
      />
      <label>
        Average
      </label>
    </div>
  </div>
  <div
    className="field"
  >
    <div
      className="ui checkbox market-checkbox"
    >
      <input
        name="low"
        onChange={[Function]}
        type="checkbox"
      />
      <label>
        Low
      </label>
    </div>
  </div>
</div>
`;
